query-engine/connectors/sql-query-connector/src/join_utils.rs

```diff
@@ -93,62 +115,78 @@ fn compute_aggr_join_one2m(
 fn compute_aggr_join_m2m(
     rf: &RelationFieldRef,
     aggregation: AggregationType,
+    filter: Option<Filter>,
     aggregator_alias: &str,
     join_alias: &str,
     previous_join: Option<&AliasedJoin>,
 ) -> AliasedJoin {
-    let relation_table = rf.as_table();
-    let model_a = rf.model();
-    let a_ids: ModelProjection = rf.model().primary_identifier().into();
-    let a_columns: Vec<_> = a_ids.as_columns().collect();
-    let b_ids: ModelProjection = rf.related_model().primary_identifier().into();
-
-    // + SELECT A.id FROM A
-    let query = Select::from_table(model_a.as_table()).columns(a_columns.clone());
+    // _ParentToChild m2m join table
+    let m2m_table = rf.as_table();
+    // _ParentToChild.Child columns
+    let m2m_child_columns = rf.related_field().m2m_columns();
+    // Child table
+    let child_model = rf.related_model();
+    // Child primary identifiers
+    let child_ids: ModelProjection = rf.related_model().primary_identifier().into();
+    // Parent primary identifiers
+    let parent_ids: ModelProjection = rf.model().primary_identifier().into();
+    // Rendered filters
+    let conditions: ConditionTree = filter
+        .map(|f| f.aliased_cond(None))
+        .unwrap_or(ConditionTree::NoCondition);
+
+    // + SELECT _ParentToChild.Child FROM Child
+    let query = Select::from_table(child_model.as_table())
+        .columns(m2m_child_columns.clone())
+        .so_that(conditions);
 
     let aggr_expr = match aggregation {
-        AggregationType::Count => count(rf.related_field().m2m_columns()),
+        AggregationType::Count => count(m2m_child_columns.clone()),
     };
 
-    // SELECT A.id,
-    // + COUNT(_AtoB.B) AS <AGGREGATOR_ALIAS>
-    // FROM A
+    // SELECT _ParentToChild.Child,
```

Comment:
Not a big deal: I'd prefer more descriptive naming of the examples - you already improved that here, thanks! - but I needed to guess that this is the id field, so for example`ChildId`. These examples are hard enough to follow (and this one is a true monstrosity), so let's make sure they're as easy to read as possible. 

On that note, is it possible to get a fully rendered example for a query here (e.g. as a top level comment)? I have a feeling that if we don't touch this code for a bit we're gonna have a truly bad time understanding it again by piecing it together.